server/core/src/miscServerFunct.cpp

```diff
@@ -2976,53 +2983,57 @@ irods::error get_script_output_single_line(
     const std::string&              script_name,
     const std::vector<std::string>& args,
     std::string&                    output ) {
-    output.clear();
-    std::stringstream exec;
+    namespace bp = boost::process;
     try {
-    exec << script_language
-         << " " << irods::get_irods_home_directory().string()
-         << "/scripts/" << script_name;
-    } catch (const irods::exception& e) {
-        irods::log(e);
-        return ERROR(-1, "failed to get irods home directory");
+        auto resolved_program = bp::search_path(script_language);
+        if (resolved_program.empty()) {
+            return ERROR(RESOLUTION_ERROR, fmt::format("Could not resolve {} to an executable", script_language));
+        }
+        auto script_path = fmt::format("{}/scripts/{}", irods::get_irods_home_directory().c_str(), script_name);
+        if (!std::filesystem::exists(script_path)) {
+            return ERROR(RESOLUTION_ERROR, fmt::format("Script file not found! {}", script_path));
+        }
+        auto script_path_and_args = args;
+        script_path_and_args.insert(script_path_and_args.begin(), script_path);
+
+        bp::ipstream program_output;
+        bp::ipstream program_stderr;
+        bp::child c(resolved_program, script_path_and_args, bp::std_out > program_output, bp::std_err > program_stderr);
+        output.clear();
+        c.wait();
+        const auto ec = c.exit_code();
```

Comment:
I think we need to log an error if `ec` is non-zero as well.

We should also check `program_stderr` before `program_output` so we capture what happened.

In both of these cases, we only need to log the error since that is what the original code did?